spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/TraceIdLoggingEnhancer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.logging;
+
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.LoggingEnhancer;
+
+import org.springframework.cloud.gcp.core.DefaultGcpProjectIdProvider;
+
+/**
+ * Adds the trace ID to the logging entry, in its correct format to be displayed in the Logs viewer.
+ *
+ * @author João André Martins
+ */
+public class TraceIdLoggingEnhancer implements LoggingEnhancer {
+
+	private static final ThreadLocal<String> traceId = new ThreadLocal<>();
+
+	public static void setCurrentTraceId(String id) {
+		if (id == null) {
+			traceId.remove();
+		}
+		else {
+			traceId.set(id);
+		}
+	}
+
+	public static String getTraceId() {
+		return traceId.get();
+	}
+
+	@Override
+	public void enhanceLogEntry(LogEntry.Builder builder) {
+		String projectId = new DefaultGcpProjectIdProvider().getProjectId();
```

Comment:
Totally agree we should be doing this because the devsite mentions this format (https://cloud.google.com/trace/docs/viewing-details#log_entries) .  But it makes me wonder why that other TraceLoggingEnhancer we were using even exists?  the "trace_id" label that other one uses isn't mentioned in that devsite link at all... Should we set both ?

Reply:
I opened a new issue in `google-cloud-java` so they fix their current implementation. https://github.com/GoogleCloudPlatform/google-cloud-java/issues/3280

Reply:
We really only need the trace ID in one place, it's not being used for anything in `labels.trace_id`.
Probably better not to add any more complexity where it's not necessary.